docs/custom-components.qmd

```diff
@@ -0,0 +1,314 @@
+---
+title: "Custom Javascript components"
+---
+
+In this post, you will learn how to create a custom element and accompanying output binding in Shiny. This is useful if you want to create an output that is not currently in Shiny for your app or if you want to create a custom output for a package you are developing.
+
+:::{.callout-note}
+The code shown here is simplified to get the point across, but before you use it in your own app, you should make sure to add error handling and other features to make it robust.
+:::
+
+# The problem
+
+You found a new table library that you really want to use in your Shiny app. Problem is, there's no wrapper for it, currently. The library is [Tabulator](https://tabulator.info/) and it's a javascript library for making tables with data.
+
+# The solution
+
+To implement a custom tabulator element for your app, you'll need to write three things:
+
+1. A javascript script that renders the element on the client side using the Tabulator library
+2. An `output_tabulator()` function for placing the element in your app's UI
+3. A `render_tabulator()` decorator for passing table data to the javascript code rendering the element on the server side
+
```

Comment:
I think it would be nice here to have a mermaid diagram of how all the components work together and what they do.